File: JavaApp/textWatcher/app/src/main/java/com/example/textwatcher/MainActivity.java
package com.example.textwatcher;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.example.textwatcher.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.et.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                binding.tv.setTextColor(Color.YELLOW);
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                binding.tv.setTextColor(Color.GREEN);
            }

            @Override
            public void afterTextChanged(Editable s) {
                binding.tv.setTextColor(Color.BLUE);
            }
        });
    }
}

File: MVC/SPIM/app/src/main/java/com/example/spim/controller/activity/MainActivity.kt
package com.example.spim.controller.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.spim.R
import com.example.spim.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }
}